rabbitmq/src/main/resources/definitions.json

```diff
@@ -33,6 +33,14 @@
     },
     {
       "vhost": "/",
+      "name": "VRO-GLOBAL-MESSAGE-TTL",
+      "pattern": ".*",
+      "apply-to": "queues",
+      "definition": {
+        "message-ttl": 1500000
```

Comment:
Every resource (queues/exchanges) can only have one policy applied at a time, and if the priorities match, then the policy applied to any queue matching the pattern (as in regex) is basically random and not guaranteed to be consistent. 

This means that you really need to add the message-ttl to every policy if that's the desired behavior.

Reply:
@dfitchett  this is not great as there is no way to enforce that for any new policies include this condition which we will come to depend on in our monitoring for uptime.

Is there a reason we don't set the more specific properties at the code level. I was thinking that we could keep this file more for global properties of rabbitmq

Reply:
> @dfitchett this is not great as there is no way to enforce that for any new policies include this condition which we will come to depend on in our monitoring for uptime.

We will have to enforce this through pull requests, and ensure that each of the desired properties are added to the appropriate policies.

> Is there a reason we don't set the more specific properties at the code level. I was thinking that we could keep this file more for global properties of rabbitmq

The main reason for setting some of these properties via policy, is **deployability**. For example, if `message-ttl` is set on the queue at declaration time for a specific queue, then every service that declares that queue for use must also set that property on declaration. If the two or more services using that queue do not declare it exactly, then only the first one to declare it will be able to connect to it, and the others will fail with a 406 (see [here](https://www.rabbitmq.com/docs/queues#property-equivalence)).

Lets say that we change all services to match the new configuration, then all those services will have to be redeployed, which we just tried to go through (painfully) with the dead letter configurations. Additionally, queues/exchanges will need to be manually deleted if any queues/exchanges were declared with `autodelete=false` or any cases where `autodelete=true` but the queue never had a consumer, or the exchange never had any bindings (meaning they will not autodelete themselves). 

The docs say that its recommended that all optional parameters are set via policy:

> Optional queue arguments can be set in a couple of ways:
> * To groups of queues using [policies](https://www.rabbitmq.com/docs/parameters#policies) (recommended)
> * On a per-queue basis when a queue is declared by a client
> * For the x-queue-type argument, [using a default queue type](https://www.rabbitmq.com/docs/vhosts#default-queue-type)
>
> The former option is more flexible, non-intrusive, does not require application modifications and redeployments. Therefore it is highly recommended for most users.

Additionally, there are a handful of properties that are better set at the message level on publish as opposed to at the policy level, `message-ttl` being on of them. By setting this at the client level, the client (publisher) is charge of how long it should wait for that message to be consumed. 

For example, the EP merge app will add a message-ttl via the `expires = 0` property to every request it makes to the BIP and BGS services. This means that if the BIP or BGS service is not available to consume the message, the message is dropped or dead lettered immediately, and EP merge assumes it failed, and will continue in processing the merge appropriately. If another application was willing to wait longer, they could set it to 60 seconds or whatever is appropriate for that application's logic.